utils/createhttpclient.go

```diff
@@ -2,14 +2,33 @@ package utils
 
 import (
 	"crypto/tls"
+	"crypto/x509"
 	"net/http"
 	"os"
 	"strings"
 )
 
 func CreateHTTPClient() *http.Client {
 	skipVerify := strings.ToLower(os.Getenv("GOTIFY_SKIP_VERIFY_TLS")) == "true"
+	certFile, hasCA := os.LookupEnv("GOTIFY_TLS_ROOT_CA")
+
 	customTransport := http.DefaultTransport.(*http.Transport).Clone()
-	customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: skipVerify}
+	rootCAs := customTransport.TLSClientConfig.RootCAs
+	if hasCA {
+		cert, err := os.ReadFile(certFile)
+		if err != nil {
+			Exit1With("Failed to read CA file:", err)
+		}
+		rootCAs = x509.NewCertPool()
+		ok := rootCAs.AppendCertsFromPEM(cert)
```

Comment:
Does this have to be a CA certificate or can I just upload the server certificate here and Go will trust it (which will automatically solve my concern on pinning certificates)?

Reply:
I just tried adding the server's certificate (signed by another CA) and it was trusted surprisingly. So I guess although the official document says it needs to be a CA cert, it can actually be used to pin a certificate.

So we don't need an additional option to pin a cert.

Reply:
Nice! :confetti_ball: